src/Field/NominatimSearch/NominatimSearch.jsx

```diff
@@ -0,0 +1,276 @@
+     * Restrict the results to only items contained with the bounding box.
+     * Restricting the results to the bounding box also enables searching by
+     * amenity only. For example a search query of just "[pub]" would normally be
+     * rejected but with bounded=1 will result in a list of items matching within
+     * the bounding box.
+     * @type {Number}
+     */
+    bounded: 1,
+
+    /**
+     * Output geometry of results in geojson format.
+     * @type {Number}
+     */
+    polygon_geojson: 1,
+
+    /**
+     * Include a breakdown of the address into elements.
+     * @type {Number}
+     */
+    addressdetails: 1,
+
+    /**
+     * Limit the number of returned results.
+     * @type {Number}
+     */
+    limit: 10,
+
+    /**
+     * Limit search results to a specific country (or a list of countries).
+     * <countrycode> should be the ISO 3166-1alpha2 code, e.g. gb for the United
+     * Kingdom, de for Germany, etc.
+     * @type {String}
+     */
+    countrycodes: 'de',
+
+    /**
+     * The minimal amount of characters entered in the input to start a search.
+     * @type {Number}
+     */
+    minChars: 3,
+
+    /**
+     * The style object passed to the AutoComplete.
+     * @type {Object}
+     */
+    style: {
+      width: 200
+    }
+  }
+
+  /**
+   * Create the NominatimSearch.
+   *
+   * @param {Object} props The initial props.
+   * @constructs NominatimSearch
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchTerm: '',
+      dataSource: []
+    };
+    this.onUpdateInput = this.onUpdateInput.bind(this);
+    this.onMenuItemSelected = this.onMenuItemSelected.bind(this);
+  }
+
+  /**
+   * Called if the input of the AutoComplete is being updated. It sets the
+   * current inputValue as searchTerm and starts a search if the inputValue has
+   * a length of at least `this.props.minChars` (default 3).
+   *
+   * @param {String|undefined} inputValue The inputValue. Undefined if clear btn
+   *                                      is pressed.
+   */
+  onUpdateInput(inputValue) {
+    this.setState({
+      dataSource: []
+    });
+
+    this.setState({
+      searchTerm: inputValue || ''
+    }, () => {
+      if (this.state.searchTerm.length >= this.props.minChars) {
+        this.doSearch();
+      }
+    });
+  }
+
+  /**
+   * Perform the search.
+   */
+  doSearch() {
+    const baseParams = {
+      format: this.props.format,
+      viewbox: this.props.viewbox,
+      bounded: this.props.bounded,
+      polygon_geojson: this.props.polygon_geojson,
+      addressdetails: this.props.addressdetails,
+      limit: this.props.limit,
+      countrycodes: this.props.countrycodes,
+      q: this.state.searchTerm
+    };
+
+    const getRequestParams = UrlUtil.objectToRequestString(baseParams);
+
+    fetch(`${this.props.nominatimBaseUrl}${getRequestParams}`)
+      .then(response => response.json())
+      .then(this.onFetchSuccess.bind(this))
+      .catch(this.onFetchError);
```

Comment:
Missing bind?

Reply:
It just logs an error and `this` is not used, so we don't need it. But i'll ad it to be consistent.